File: Assignment8/C/Program5.c
// PS - Write a program which acepts one number from user and return diff between sumation of even digit and summation of odd digits

#include<stdio.h>

int Difference(int iNo) {
    int iDigit = 0;
    int iEven = 0;
    int iOdd = 0;


    if(iNo < 0) {
        iNo = -iNo;
    }

    while(iNo != 0) {
        iDigit = iNo % 10;
        if((iDigit % 2) == 0) {
            iEven = iEven + iDigit;
        }
        if((iDigit % 2) != 0) {
            iOdd = iOdd + iDigit;
        }
        iNo = iNo / 10;
    }

    return (iEven - iOdd);
}

int main() {
    int iValue = 0;
    int iRet = 0;

    printf("Enter number \n");
    scanf("%d", &iValue);

    iRet = Difference(iValue);

    printf("Difference between sumation of even digit and summation of odd digits %d \n", iRet);

    return 0;
}

File: Assignment35/Java/Program5.java
 // PS - Write a java program which accept number from user and return Difference of summation of even and odd digits

import java.util.*;

class DigitDemo
{
    public int SumEven(int No)
    {
        int iSum = 0;
        int iDigit = 0;

        while(No != 0)
        {
            iDigit = No % 10;
            if((iDigit % 2) == 0)
            {
                iSum = iSum + iDigit;
            }
            No = No / 10;
        }
        return iSum;
    }

    public int SumOdd(int No)
    {
        int iSum = 0;
        int iDigit = 0;

        while(No != 0)
        {
            iDigit = No % 10;
            if((iDigit % 2) != 0)
            {
                iSum = iSum + iDigit;
            }
            No = No / 10;
        }
        return iSum;
    }

    public int Difference(int No)
    {
        int Even = SumEven(No);
        int Odd = SumOdd(No);

        return Even - Odd;
    }
}

class Program5
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number");
        int No = sobj.nextInt();

        DigitDemo dobj = new DigitDemo();
        int iRet = dobj.Difference(No);

        System.out.println("Difference of summation of even and odd digits are : "+iRet);
    }
}